subprojects/parseq/src/test/java/com/linkedin/parseq/AbstractTaskTest.java

```diff
@@ -297,6 +297,29 @@ public void testWithSideEffectFailure(int expectedNumberOfTasks) throws Exceptio
     assertEquals(countTasks(failure.getTrace()), expectedNumberOfTasks);
   }
 
+  @Test
+  public void testStaticWithSideEffectFullCompletion() throws Exception {
+    Task<String> slowSideEffect = delayedValue("slow", 50, TimeUnit.MILLISECONDS);
+    Task<String> sideEffect = Task.withSideEffect(() -> slowSideEffect);
+
+    run(sideEffect);
+    assertFalse(sideEffect.isDone());
+    sideEffect.await();
+    assertTrue(sideEffect.isDone());
+    assertNull(sideEffect.get());
+  }
+
+  @Test
+  public void testStaticSideEffectFailureIsIgnored() throws Exception {
```

Comment:
Maybe  also add a test in `TestPlanCompletionListener` to make sure that plan completion listener is also properly triggered for this usage.